### PR TITLE
Components: Introducing `SearchField`

### PR DESCRIPTION
**Pull request by OsamaAbdellateef**

### CODE REVIEW

**Comment by sebald** [comment]
Needs docs, wait for the new docs until we merge this?

**Comment by OsamaAbdellateef** [comment]
> Needs docs, wait for the new docs until we merge this?

Yup could be in the new version of our doc

**Comment by OsamaAbdellateef** [comment]
> Needs docs, wait for the new docs until we merge this?

@sebald 
now it is ready

File: packages/components/src/SearchField/SearchField.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    description: 'This is a help text description',
+    errorMessage: 'Something went wrong',
+  },
+} satisfies Meta<typeof SearchField>;
+
+export default meta;
+type Story = StoryObj<typeof meta>;
+
+export const Basic: Story = {
+  render: args => <SearchField required label="search field" />,
```
Comment (sarahgm) [inline packages/components/src/SearchField/SearchField.stories.tsx:54]: forgot to add args in there

**Review by sarahgm — APPROVED** [review]

**Pull request merged**